segments/cwd.py

```diff
@@ -21,17 +21,21 @@ def add_cwd_segment():
     if len(names) > max_depth:
         names = names[:2] + [u'\u2026'] + names[2 - max_depth:]
 
-    if not powerline.args.cwd_only:
-        for n in names[:-1]:
-            if n == '~' and Color.HOME_SPECIAL_DISPLAY:
-                powerline.append(' %s ' % n, Color.HOME_FG, Color.HOME_BG)
-            else:
-                powerline.append(' %s ' % n, Color.PATH_FG, Color.PATH_BG,
-                    powerline.separator_thin, Color.SEPARATOR_FG)
-
-    if names[-1] == '~' and Color.HOME_SPECIAL_DISPLAY:
-        powerline.append(' %s ' % names[-1], Color.HOME_FG, Color.HOME_BG)
+    if powerline.args.cwd_mode=='plain':
+        powerline.append(' %s%s ' % ('' if names[0] in '/~' else '/', '/'.join(names)), 
```

Comment:
To be clear, I am not asking you to do that refactor. If you just switch this code to use `cwd` I would be cool to merge. I can tackle the refactor after merging.